src/Plugin/EdgeKeyTypeBase.php

```diff
@@ -58,8 +58,11 @@ public function getAuthenticationType(KeyInterface $key): string {
   /**
    * {@inheritdoc}
    */
-  public function getEndpoint(KeyInterface $key): string {
-    return $key->getKeyValues()['endpoint'] ?? Client::DEFAULT_ENDPOINT;
+  public function getEndpoint(KeyInterface $key, bool $safeReturn = FALSE): ?string {
+    if ($safeReturn) {
+      return $key->getKeyValues()['endpoint'] ?? Client::DEFAULT_ENDPOINT;
```

Comment:
Retrieve the value from the interface instead of the object. `\Apigee\Edge\ClientInterface::DEFAULT_ENDPOINT`